src/app/[locale]/(other)/admin/notifications/page.tsx

```diff
@@ -0,0 +1,431 @@
+                                <SelectTrigger>
+                                    <SelectValue placeholder="All statuses" />
+                                </SelectTrigger>
+                                <SelectContent>
+                                    <SelectItem value="all">All statuses</SelectItem>
+                                    <SelectItem value="pending">Pending</SelectItem>
+                                    <SelectItem value="sent">Sent</SelectItem>
+                                    <SelectItem value="failed">Failed</SelectItem>
+                                </SelectContent>
+                            </Select>
+                        </div>
+
+                        <div>
+                            <label className="text-sm font-medium mb-2 block">Type</label>
+                            <Select value={filters.type} onValueChange={(value) => setFilters({ ...filters, type: value })}>
+                                <SelectTrigger>
+                                    <SelectValue placeholder="All types" />
+                                </SelectTrigger>
+                                <SelectContent>
+                                    <SelectItem value="all">All types</SelectItem>
+                                    <SelectItem value="beforeMeeting">Before Meeting</SelectItem>
+                                    <SelectItem value="afterMeeting">After Meeting</SelectItem>
+                                </SelectContent>
+                            </Select>
+                        </div>
+
+                        <div className="flex items-end">
+                            <Button onClick={fetchNotifications} variant="outline" className="w-full">
+                                Apply Filters
+                            </Button>
+                        </div>
+                    </div>
+                </CardContent>
+            </Card>
+
+            {/* Notifications List */}
+            {loading ? (
+                <div className="flex justify-center py-12">
+                    <Loader2 className="h-8 w-8 animate-spin text-gray-400" />
+                </div>
+            ) : Object.keys(groupedNotifications).length === 0 ? (
+                <Card>
+                    <CardContent className="flex flex-col items-center justify-center py-12">
+                        <Bell className="h-12 w-12 text-gray-400 mb-4" />
+                        <p className="text-lg font-medium text-gray-900">No notifications found</p>
+                        <p className="text-sm text-gray-500">Try adjusting your filters</p>
+                    </CardContent>
+                </Card>
+            ) : (
+                <div className="space-y-4">
+                    {Object.values(groupedNotifications).map((group: any) => {
+                        const pendingCount = getPendingCount(group.notifications);
+                        const pendingIds = getPendingNotificationIds(group.notifications);
```

Comment:
Computing `pendingCount` and `pendingIds` inside the .map() callback creates new values on every render. `pendingIds` is a new array reference each time and is captured in the `onClick` closure on line 266. This can cause unnecessary recalculations on every render.

Consider memoizing these values (e.g., with useMemo) or computing them outside the render path to ensure stable references.